packages/nvidia_nat_mcp/src/nat/plugins/mcp/client_impl.py

```
@@ -179,6 +179,13 @@ async def mcp_client_function_group(config: MCPClientConfig, _builder: Builder):
     group = FunctionGroup(config=config)
 
     async with client:
+        # Expose the live MCP client on the function group instance so other components (e.g., HTTP endpoints)
+        # can reuse the already-established session instead of creating a new client per request.
+        # These attributes are intentionally internal/private to avoid API surface commitments.
+        setattr(group, "_mcp_client", client)
```

Ruff (0.13.1)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)